templates/FFV_form_factors.cpp.in

```diff
@@ -425,4 +617,364 @@ std::valarray<std::complex<double>> FFV_FFS<Fj, Fi, V, FA, FB, S>::value(
+   // static_assert(
+   //     std::is_same<VA, GB>::type::value,
+   //     "Internal vectors in the FFV_VGF instantiation must be of the same type."
+   // );
+
+   // use P for external massless vector so as not to confuse with the internal vectors
+   using VertexFBarFjVBar = Vertex<typename F::lorentz_conjugate, typename VA::lorentz_conjugate, Fj>;
+   using VertexFiBarFG = Vertex<typename Fi::lorentz_conjugate, GB, F>;
+   using VertexGBarVPBar = Vertex<typename GB::lorentz_conjugate, VA, typename P::lorentz_conjugate>;
+
+   // masses of external fermions
+   const auto mj = context.mass<Fj>(indices_in);
+   const auto mi = context.mass<Fi>(indices_out);
+
+   // loop over all possible particle generations attached to both vertices
+   std::valarray<std::complex<double>> res {0.0, 0.0, 0.0, 0.0};
+   for (const auto& indexIn: index_range<VertexFBarFjVBar>()) {
+      for (const auto& indexOut: index_range<VertexFiBarFG>()) {
+          
+         // cycle if generations of external fermions are different then requested
+         const auto jFieldIndices = VertexFBarFjVBar::template indices_of_field<2>(indexIn);
+         const auto iFieldIndices = VertexFiBarFG::template indices_of_field<0>(indexOut);
+         if (jFieldIndices != indices_in || iFieldIndices != indices_out)
+            continue;
+         
+         // match indices of the fermion in the loop
+         const auto fermionFieldIndicesIn = VertexFBarFjVBar::template indices_of_field<0>(indexIn);
+         const auto fermionFieldIndicesOut = VertexFiBarFG::template indices_of_field<2>(indexOut);
+         if (fermionFieldIndicesIn != fermionFieldIndicesOut)
+            continue;
+
+         // cannot match indices if scalar is not would-be-goldstone corresponding to vector
+         const auto vectorFieldIndicesIn = VertexFBarFjVBar::template indices_of_field<1>(indexIn);
+         const auto goldstoneFieldIndicesOut = VertexFiBarFG::template indices_of_field<1>(indexOut);
+         // if (vectorFieldIndicesIn != goldstoneFieldIndicesOut)
+         //   continue;
+
+         // ignore contribution if both vector and fermion are SM particles
+         if (discard_SM_contributions) {
+            if (isSMField<VA>(vectorFieldIndicesIn) && isSMField<GB>(goldstoneFieldIndicesOut) && isSMField<F>(fermionFieldIndicesIn)) {
+               continue;
+            }
+         }
+
+         const auto vertexIn = VertexFBarFjVBar::evaluate(indexIn, context);
+         const auto vertexOut = VertexFiBarFG::evaluate(indexOut, context);
+
+         const auto indexEmit = concatenate(vectorFieldIndicesIn, goldstoneFieldIndicesOut);
+         const auto vertexEmit = VertexGBarVPBar::evaluate(indexEmit, context);
+
+         const auto mV = context.mass<VA>(vectorFieldIndicesIn);
+         const auto mG = context.mass<GB>(goldstoneFieldIndicesOut);
+         const auto mF = context.mass<F>(fermionFieldIndicesIn);
+         const auto x = pow(mF/mV, 2);
```

Comment:
See my comment to the loop functions regarding the `pow()` function. You may use the `Power2()` function here.